fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -272,6 +278,10 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
                 navHost.navController.navigate(NavGraphDirections.actionGlobalHomeJunoOnboarding())
             }
         } else {
+            lifecycleScope.launch(IO) {
+                showFullscreenMessageSurveyIfNeeded(applicationContext)
```

Comment:
No sure if this is the expected behaviour, we will check once, if we haven't loaded the messages before this point we my not show the message.

I'm not sure if should reuse what we have for the home screen message, by dispatching a new action for this new surface.

https://github.com/mozilla-mobile/firefox-android/blob/865b1d75f1376d043538882b6001863f7a22f980/fenix/app/src/main/java/org/mozilla/fenix/messaging/MessagingFeature.kt#L17

and observe [messageToShow](https://github.com/mozilla-mobile/firefox-android/blob/865b1d75f1376d043538882b6001863f7a22f980/fenix/app/src/main/java/org/mozilla/fenix/components/appstate/AppState.kt#L70) changes?

That will warranty that the message will show whenever we re enter the Home Screen, but I'm not sure if this is what we want?

Reply:
That is a good idea but I don't think it's for this case.  Homescreen messages are handled differently because it's part of the session adapter.  In this case we want to navigate to the full screen message directly.

Reply:
👍🏽